src/librustc/middle/mem_categorization.rs

```diff
@@ -1205,13 +1205,16 @@ impl<'t,'tcx,TYPER:Typer<'tcx>> MemCategorizationContext<'t,TYPER> {
         // Note: This goes up here (rather than within the PatEnum arm
         // alone) because struct patterns can refer to struct types or
         // to struct variants within enums.
-        let cmt = match opt_def {
-            Some(&def::DefVariant(enum_did, variant_did, _))
-                // univariant enums do not need downcasts
-                if !ty::enum_is_univariant(self.tcx(), enum_did) => {
-                    self.cat_downcast(pat, cmt.clone(), cmt.ty, variant_did)
-                }
-            _ => cmt
+        let cmt = if let Some(&def::DefVariant(enum_did, variant_did, _)) = opt_def {
+            let enum_def = ty::lookup_datatype_def(self.tcx(), enum_did);
+            // Univariant enums don't need a downcast
+            if !enum_def.is_univariant() {
+                self.cat_downcast(pat, cmt.clone(), cmt.ty, variant_did)
```

Comment:
Down... cast? This is pre-existing, but what does it actually mean? Some sort of ancient refinement relic?